eng/helix/content/RunTests/TestRunner.cs

```diff
@@ -213,50 +213,60 @@ public async Task<int> RunTestsAsync()
             try
             {
                 // Timeout test run 5 minutes before the Helix job would timeout
-                var cts = new CancellationTokenSource(Options.Timeout.Subtract(TimeSpan.FromMinutes(5)));
+                var testProcessTimeout = Options.Timeout.Subtract(TimeSpan.FromMinutes(5));
+                var cts = new CancellationTokenSource(testProcessTimeout);
                 var diagLog = Path.Combine(Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT"), "vstest.log");
-                var commonTestArgs = $"test {Options.Target} --diag:{diagLog} --logger:xunit --logger:\"console;verbosity=normal\" --blame \"CollectHangDump;TestTimeout=15m\"";
+                var commonTestArgs = $"test {Options.Target} --diag:{diagLog} --logger:xunit --logger:\"console;verbosity=normal\" --blame \"CollectHangDump;CollectDump;TestTimeout=15m\"";
                 if (Options.Quarantined)
                 {
-                    Console.WriteLine("Running quarantined tests.");
+                    ProcessUtil.PrintMessage("Running quarantined tests.");
 
                     // Filter syntax: https://github.com/Microsoft/vstest-docs/blob/master/docs/filter.md
                     var result = await ProcessUtil.RunAsync($"{Options.DotnetRoot}/dotnet",
                         commonTestArgs + " --TestCaseFilter:\"Quarantined=true\"",
                         environmentVariables: EnvironmentVariables,
-                        outputDataReceived: Console.WriteLine,
-                        errorDataReceived: Console.Error.WriteLine,
+                        outputDataReceived: ProcessUtil.PrintMessage,
+                        errorDataReceived: ProcessUtil.PrintErrorMessage,
                         throwOnError: false,
                         cancellationToken: cts.Token);
 
                     if (result.ExitCode != 0)
                     {
-                        Console.WriteLine($"Failure in quarantined tests. Exit code: {result.ExitCode}.");
+                        if (cts.Token.IsCancellationRequested)
```

Comment:
I think this is also missing the change that moved this outside of the exit code check: https://github.com/dotnet/aspnetcore/commit/cab4c6ec6e900b2c4b5b93f2237c6972efebc02a#diff-bf8c10d051da7b9b3b26d56a968fe4ea30048d1da46a7474c75dd089c50bc1ffR241